README.md

```diff
@@ -28,54 +40,174 @@ A hotel booking application in React. Homework for the [CodeYourFuture React mod
+
+**Test:** Clicking the button should still increment the number of pizzas.
+
+#### 12. Extract pizza order in its own Order component
+
+**Instructions:** Extract the `<li>` containing "Pizzas" from the `<Restaurant />` component to a new component named `Order`. Also, move the declaration of the `orders` state and the `orderOne` function from the `<Restaurant />` component to the new `<Order />` component. Use the `<Order />` component in the `<ul>` list of the `<Restaurant />` component.
+
+**Test:** Make sure the pizza order is still rendered on the page and that clicking on the "Add" button still increments the number of orders.
+
+#### 13. Render more orders
+
+**Instructions:** Pass a new prop named `orderType` to the `<Order />` component with the value "Pizzas". Then render the `orderType` prop instead of "Pizzas" in the `<Order />` component. Make sure that "Pizzas" is still displayed on the screen. In the `<ul>` list of the `<Restaurant />` component, render 2 others `<Order />` components but this time pass different values for the `orderType` prop: "Salads" and "Chocolate cake".
+
+**Test:** For each order, the number of items can be incremented independently. Verify that you are able to explain what is happening.
+
+#### 14. Holding passing bookings from a state variable
+
+**Instructions:** In the `<Bookings />` component, declare a new state `bookings` with the corresponding setter function `setBookings` to hold the `FakeBookings` data. Instead of passing `FakeBookings` directly to the `<SearchResults />` component, pass the new `bookings` state variable.
+
+**Hint:** The new `bookings` state should be initialised with the `FakeBookings` variable.
+
+**Test:** Check that the bookings are still rendered correctly in the page.
+
+#### 15. Load bookings remotely
+
+**Instructions:** Instead of getting the existing bookings from the file `data/fakeBookings.json`, we will get and load the bookings from a remote API. In the `<Bookings />` component, use the React function `useEffect` to `console.log()` some text only when the page first renders on the screen. Verify that when you refresh the page, the text appears once in the console. Then, in the `useEffect` function, use the `fetch()` function to get data from `https://cyf-react.glitch.me`.
+
+**Hints:**
+
+- Replace `FakeBookings` in the bookings state and initialise it with `[]` (because we haven't fetched any results yet!)
+- After calling the `fetch()` function, use `.then()` to handle the response. Try looking at your Pokemon app that you worked on in class for an example
+- When the response comes back, use `setBookings` to update the results
+
+**Test:** Verify the customers data are still displayed correctly in the table.
+
+#### 16. Highlight booking row when clicked
```

Comment:
Historically, this has been the hardest exercise (I think most students haven't been able to complete it). So I think I would suggest making it a stretch goal.

On the other hand I think it's also the exercise which *teaches* the most: it really get students to "think in components", because the simplest solution is to push the selected state down to individual rows in the table. Most students approach it by trying to model multiple selection states in the parent.

Reply:
I also think it's an interesting exercise, it appears as the last task of the homework for lesson 2, that should be okay i think? I'm going to introduce a button as you suggested for the task 21, so this task 16 is not a blocker if students don't get it 100% right.

Reply:
Sounds good, thanks.

The other advantage of it being in week 2, is that we can explain the solution at the beginning of the week 3 lesson.